### ISSUE TITLE
arduino-ide-1.6.8 fails on missing org.apache.commons.net.util.SubnetUtils

### ISSUE THREAD

**Issue by mkyral** [maurerpe/arduino-overlay#2]
Hi,
I'm trying to build arduino-ide on my stable system and it fails on missing org.apache.commons.net.util.SubnetUtils library. What I need to install?

```
>>> Emerging (1 of 1) dev-embedded/arduino-ide-1.6.8::arduino-overlay
 * 1.6.8.tar.gz SHA256 SHA512 WHIRLPOOL size ;-) ...                                                                                                                                                                              [ ok ]
 * Using: icedtea-bin-8
>>> Unpacking source...
>>> Unpacking 1.6.8.tar.gz to /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work
>>> Source unpacked in /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work
>>> Preparing source in /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8 ...
 * Applying arduino-ide-1.6.8-build.xml.patch ...                                                                                                                                                                                 [ ok ]
 * Applying arduino-ide-1.6.8-startup.patch ...                                                                                                                                                                                   [ ok ]
>>> Source prepared.
>>> Configuring source in /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8 ...
Rewriting attributes
Rewriting ./build/build.xml
Rewriting ./arduino-core/build.xml
Rewriting ./app/build.xml
>>> Source configured.
>>> Compiling source in /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8 ...
 * Disabling all optional ANT_TASKS
Buildfile: /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/build.xml

build:

package-library-index-json-bundle:
    [mkdir] Created dir: /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist
      [get] Getting: http://downloads.arduino.cc/packages/package_index.json.gz
      [get] To: /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/package_index.json.gz
      [get] .
   [gunzip] Expanding /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/package_index.json.gz to /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/package_index.json
   [delete] Deleting: /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/package_index.json.gz
      [get] Getting: http://downloads.arduino.cc/packages/package_index.json.sig
      [get] To: /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/package_index.json.sig
      [get] .
      [get] Getting: http://downloads.arduino.cc/libraries/library_index.json.gz
      [get] To: /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/library_index.json.gz
      [get] ..........................
   [gunzip] Expanding /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/library_index.json.gz to /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/library_index.json
   [delete] Deleting: /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/linux/work/dist/library_index.json.gz

revision-check:
     [echo] AVR Arduino core version in platform.txt is: 1.6.10
     [echo]                     latest in index.json is: 1.6.10
     [echo] Latest revision detected in shared/revision.txt is: 1.6.8
     [echo]                      Revision in BaseNoGui.java is: 1.6.8

linux-checkos:

subprojects-build:

compile:
    [mkdir] Created dir: /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/arduino-core/bin
    [javac] Compiling 135 source files to /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/arduino-core/bin
    [javac] /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/arduino-core/src/cc/arduino/net/PACSupportMethods.java:33: error: cannot find symbol
    [javac] import org.apache.commons.net.util.SubnetUtils;
    [javac]                                   ^
    [javac]   symbol:   class SubnetUtils
    [javac]   location: package org.apache.commons.net.util
    [javac] /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/arduino-core/src/cc/arduino/net/PACSupportMethods.java:67: error: cannot find symbol
    [javac]     SubnetUtils subnetUtils = new SubnetUtils(pattern, mask);
    [javac]     ^
    [javac]   symbol:   class SubnetUtils
    [javac]   location: class PACSupportMethods
    [javac] /var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/arduino-core/src/cc/arduino/net/PACSupportMethods.java:67: error: cannot find symbol
    [javac]     SubnetUtils subnetUtils = new SubnetUtils(pattern, mask);
    [javac]                                   ^
    [javac]   symbol:   class SubnetUtils
    [javac]   location: class PACSupportMethods
    [javac] 3 errors

BUILD FAILED
/var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/build.xml:91: The following error occurred while executing this line:
/var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/build/build.xml:138: The following error occurred while executing this line:
/var/tmp/portage/dev-embedded/arduino-ide-1.6.8/work/Arduino-1.6.8/arduino-core/build.xml:26: Compile failed; see the compiler error output for details.
```

```
# emerge --info
Portage 2.2.26 (python 3.4.3-final-0, default/linux/amd64/13.0/desktop/plasma, gcc-4.9.3, glibc-2.22-r4, 4.4.5-gentoo x86_64)
=================================================================
System uname: Linux-4.4.5-gentoo-x86_64-Intel-R-_Core-TM-_i7-4700MQ_CPU_@_2.40GHz-with-gentoo-2.2
KiB Mem:    16341816 total,   8394796 free
KiB Swap:    8388604 total,   8388604 free
Timestamp of repository gentoo: Fri, 15 Apr 2016 12:03:23 +0000
sh bash 4.3_p42-r1
ld GNU ld (Gentoo 2.25.1 p1.1) 2.25.1
app-shells/bash:          4.3_p42-r1::gentoo
dev-java/java-config:     2.2.0-r3::gentoo
dev-lang/perl:            5.20.2::gentoo
dev-lang/python:          2.7.10-r1::gentoo, 3.4.3-r1::gentoo, 3.5.0-r1::gentoo
dev-util/cmake:           3.3.1-r1::gentoo
dev-util/pkgconfig:       0.28-r2::gentoo
sys-apps/baselayout:      2.2::gentoo
sys-apps/openrc:          0.19.1::gentoo
sys-apps/sandbox:         2.10-r1::gentoo
sys-devel/autoconf:       2.13::gentoo, 2.69::gentoo
sys-devel/automake:       1.11.6-r1::gentoo, 1.14.1::gentoo, 1.15::gentoo
sys-devel/binutils:       2.25.1-r1::gentoo
sys-devel/gcc:            4.9.3::gentoo
sys-devel/gcc-config:     1.7.3::gentoo
sys-devel/libtool:        2.4.6::gentoo
sys-devel/make:           4.1-r1::gentoo
sys-kernel/linux-headers: 4.5::gentoo (virtual/os-headers)
sys-libs/glibc:           2.22-r4::gentoo
Repositories:

gentoo
    location: /mnt/dba/portage/tree
    sync-type: git
    sync-uri: https://github.com/gentoo-mirror/gentoo
    priority: -1000

arduino-overlay
    location: /var/lib/portage-overlays/arduino
    sync-type: git
    sync-uri: https://github.com/maurerpe/arduino-overlay
    masters: gentoo

grub2-themes
    location: /var/lib/portage-overlays/grub2-themes
    sync-type: git
    sync-uri: https://github.com/gentoo/grub2-themes-overlay.git
    masters: gentoo

java
    location: /var/lib/portage-overlays/java
    sync-type: git
    sync-uri: git://anongit.gentoo.org/proj/java.git
    masters: gentoo

kde
    location: /var/lib/portage-overlays/kde
    sync-type: git
    sync-uri: git://anongit.gentoo.org/proj/kde.git
    masters: gentoo

mkyral
    location: /usr/local/portage
    masters: gentoo
    priority: 0

python
    location: /var/lib/portage-overlays/python
    sync-type: git
    sync-uri: git://anongit.gentoo.org/proj/python.git
    masters: gentoo

seden
    location: /var/lib/portage-overlays/seden
    sync-type: git
    sync-uri: git://anongit.gentoo.org/user/seden.git
    masters: gentoo

sunrise
    location: /var/lib/portage-overlays/sunrise
    sync-type: git
    sync-uri: git://anongit.gentoo.org/proj/sunrise-reviewed.git
    masters: gentoo

ACCEPT_KEYWORDS="amd64"
ACCEPT_LICENSE="* -@EULA"
CBUILD="x86_64-pc-linux-gnu"
CFLAGS="-march=native -O2 -pipe"
CHOST="x86_64-pc-linux-gnu"
CONFIG_PROTECT="/etc /usr/lib64/libreoffice/program/sofficerc /usr/share/config /usr/share/gnupg/qualified.txt /var/lib/hsqldb"
CONFIG_PROTECT_MASK="/etc/ca-certificates.conf /etc/dconf /etc/env.d /etc/fonts/fonts.conf /etc/gconf /etc/gentoo-release /etc/php/apache2-php5.6/ext-active/ /etc/php/cgi-php5.6/ext-active/ /etc/php/cli-php5.6/ext-active/ /etc/revdep-rebuild /etc/sandbox.d /etc/terminfo"
CXXFLAGS="-march=native -O2 -pipe"
DISTDIR="/mnt/dba/portage/distfiles"
EMERGE_DEFAULT_OPTS="--keep-going --with-bdeps=y --backtrack=30"
FCFLAGS="-O2 -pipe"
FEATURES="assume-digests binpkg-logs config-protect-if-modified distlocks ebuild-locks fixlafiles merge-sync news parallel-fetch preserve-libs protect-owned sandbox sfperms splitdebug strict unknown-features-warn unmerge-logs unmerge-orphans userfetch userpriv usersandbox usersync xattr"
FFLAGS="-O2 -pipe"
GENTOO_MIRRORS="http://gentoo.supp.name/ http://ftp.fi.muni.cz/pub/linux/gentoo/ rsync://ftp.fi.muni.cz/pub/linux/gentoo/ ftp://ftp.fi.muni.cz/pub/linux/gentoo/ http://gentoo.mirror.dkm.cz/pub/gentoo/ ftp://gentoo.mirror.dkm.cz/pub/gentoo/ rsync://gentoo.mirror.dkm.cz/gentoo/ http://gentoo.mirror.web4u.cz/ ftp://gentoo.mirror.web4u.cz/"
LANG="cs_CZ.utf8"
LDFLAGS="-Wl,-O1 -Wl,--as-needed"
MAKEOPTS="-j9"
PKGDIR="/mnt/dba/portage/packages"
PORTAGE_CONFIGROOT="/"
PORTAGE_RSYNC_EXTRA_OPTS="--info=progress2,name0,del0"
PORTAGE_RSYNC_OPTS="--recursive --links --safe-links --perms --times --omit-dir-times --compress --force --whole-file --delete --stats --human-readable --timeout=180 --exclude=/distfiles --exclude=/local --exclude=/packages"
PORTAGE_TMPDIR="/var/tmp"
USE="X a52 aac acl acpi alsa amd64 berkdb bindist bluetooth boost branding bzip2 cairo cdda cdr cli consolekit cracklib crypt cups curl cxx dbus declarative dri dts dvd dvdr emboss encode exif fam firefox flac fortran gdbm gif git glamor gnutls gpm gstreamer gtk3 iconv icu imagemagick ipv6 jpeg kde kipi lcms libnotify mad mmx mmxext mng modules mp3 mp4 mpeg mplayer multilib ncurses nls nptl nsplugin ogg opengl openmp pam pango pcre pdf phonon plasma png policykit ppds python qml qt3support qt4 qt5 raw readline sdl seccomp semantic-desktop session smp spell sqlite sse sse2 ssl startup-notification subversion svg tcpd thumbnail tiff truetype udev udisks unicode upower usb vorbis widgets wxwidgets x264 xattr xcb xcomposite xinerama xml xpm xscreensaver xv xvid zlib" ABI_X86="64" ALSA_CARDS="hda-intel" APACHE2_MODULES="authn_core authz_core socache_shmcb unixd actions alias auth_basic authn_alias authn_anon authn_dbm authn_default authn_file authz_dbm authz_default authz_groupfile authz_host authz_owner authz_user autoindex cache cgi cgid dav dav_fs dav_lock deflate dir disk_cache env expires ext_filter file_cache filter headers include info log_config logio mem_cache mime mime_magic negotiation rewrite setenvif speling status unique_id userdir usertrack vhost_alias" CALLIGRA_FEATURES="kexi words flow plan sheets stage tables krita karbon braindump author" CAMERAS="ptp2" COLLECTD_PLUGINS="df interface irq load memory rrdtool swap syslog" CPU_FLAGS_X86="aes avx avx2 fma3 mmx mmxext popcnt sse sse2 sse3 sse4_1 sse4_2 ssse3" ELIBC="glibc" GPSD_PROTOCOLS="ashtech aivdm earthmate evermore fv18 garmin garmintxt gpsclock itrax mtk3301 nmea ntrip navcom oceanserver oldstyle oncore rtcm104v2 rtcm104v3 sirf superstar2 timing tsip tripmate tnt ublox ubx" GRUB_PLATFORMS="pc" INPUT_DEVICES="evdev keyboard mouse synaptics" KERNEL="linux" LCD_DEVICES="bayrad cfontz cfontz633 glk hd44780 lb216 lcdm001 mtxorb ncurses text" LIBREOFFICE_EXTENSIONS="presenter-console presenter-minimizer" LINGUAS="cs" OFFICE_IMPLEMENTATION="libreoffice" PHP_TARGETS="php5-5" PYTHON_SINGLE_TARGET="python3_4" PYTHON_TARGETS="python2_7 python3_4" QEMU_SOFTMMU_TARGETS="x86_64 arm" QEMU_USER_TARGETS="x86_64 arm" RUBY_TARGETS="ruby21" USERLAND="GNU" VIDEO_CARDS="nvidia intel modesetting" XTABLES_ADDONS="quota2 psd pknock lscan length2 ipv4options ipset ipp2p iface geoip fuzzy condition tee tarpit sysrq steal rawnat logmark ipmark dhcpmac delude chaos account"
Unset:  CC, CPPFLAGS, CTARGET, CXX, INSTALL_MASK, LC_ALL, PORTAGE_BUNZIP2_COMMAND, PORTAGE_COMPRESS, PORTAGE_COMPRESS_FLAGS, USE_PYTHON
```

**Comment by maurerpe** [comment]
That should be provided by dev-java/commons-net, which is a hard dependency of arduino-ide.  I have version 3.2 installed.

$ equery list -p -o commons-net
  Searching for commons-net ...
[-P-] [  ] dev-java/commons-net-1.4.1-r1:0
[IP-] [  ] dev-java/commons-net-3.2:0

$ equery files commons-net
  Searching for commons-net ...
  Contents of dev-java/commons-net-3.2:
/usr
/usr/share
/usr/share/commons-net
/usr/share/commons-net/lib
/usr/share/commons-net/lib/commons-net.jar
/usr/share/commons-net/package.env

$ unzip -l /usr/share/commons-net/lib/commons-net.jar | grep Subnet
     3099  03-13-2016 01:06   examples/cidr/SubnetUtilsExample.class
      237  03-13-2016 01:06   org/apache/commons/net/util/SubnetUtils$1.class
     4152  03-13-2016 01:06   org/apache/commons/net/util/SubnetUtils$SubnetInfo.class
     5150  03-13-2016 01:06   org/apache/commons/net/util/SubnetUtils.class

**Comment by mkyral** [comment]
Thanks. I've 1.4.1 installed. Upgrade to 3.2 fixed it.

```
[I] dev-java/commons-net
     Available versions:  1.4.1-r1^t ~3.2^t {doc examples source ELIBC="FreeBSD"}
     Installed versions:  1.4.1-r1^t(14:31:44 15.4.2016)(-doc -examples -source ELIBC="-FreeBSD")
     Homepage:            http://commons.apache.org/net/
     Description:         The purpose of the library is to provide fundamental protocol access, not higher-level abstractions
```

**Comment by maurerpe** [comment]
Good.  Glad that worked.  I have updated the ebuild to require a minimum version of 3.2 so others won't have the same issue.
